admin-portal/src/app/_components/EditCard.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from "axios";
+
+interface EditCardProps {
+    id: string;
+    handleCloseClick: () => void;
+}
+
+export default function EditCard({ id, handleCloseClick }: EditCardProps) {
+  // Variables to store the updated event details
+  const [updatedName, setUpdatedName] = useState<string>("");
+  const [updatedDate, setUpdatedDate] = useState<Date>(new Date());
+  const [updatedDescription, setUpdatedDescription] = useState<string>("");
+  const [updatedTags, setUpdatedTags] = useState<string[]>([]);
+    
+    // Get the event details by ID
+    const getEventById = async () => {
+        try {
+            const response: AxiosResponse = await axios.get(`http://${process.env.IP_ADDRESS}:${process.env.PORT}/events/getEventById/${id}`);
+            console.log("got event");
+            return response.data;
+        } catch (err) {
+            console.log(err);
+            return err;
+        }
+    };
+    
+    // Fetch the event details by ID
+    useEffect(() => {
+        const fetchData = async () => {
+            const data = await getEventById();
+            setUpdatedName(data.name);
+            setUpdatedDate(new Date(data.date));
+            setUpdatedDescription(data.description);
+            setUpdatedTags(data.tags || []);
+        };
+        fetchData();
+    }, []);    
+
+    // Edit the event details
+    const editEvent = async () => {
+        try {
+            const response : AxiosResponse = await axios.patch(`http://${process.env.IP_ADDRESS}:${process.env.PORT}/events/editEventDetails/${id}`,
+                {
+                    name: updatedName,
+                    date: updatedDate, 
+                    description: updatedDescription,
+                    tags: updatedTags
+                });
+            return response;
+        } catch (err) {
+            console.log(err);
+            return err;
+        }
+    }
+
+    const handleSubmit = async () => {
+        try {
+            await editEvent();
+            // Inherit the function from the parent component to close the modal
+            handleCloseClick();
+        } catch (error) {
+            console.error(error);
+        }
+    }
```

Comment:
For this section, every time the form submits, it causes a refresh which would disrupt the user experience and subsequently performs unnecessary additional network calls to get the latest event values from the refresh. 

We want to avoid the extra refresh and latency if possible to make the experience smooth; since this is an event handler, we can use 'preventDefault()' on the event object to prevent the refresh, but we'd also like to update the local information in the EventCard too!

Please implement this change so changes reflected in the DB also reflect in the front-end WITHOUT requiring a refresh.

Reply:
Same thing goes for delete as well.